tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/IComponentTemplateOperation.java

```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+
+package com.liferay.ide.project.core.modules;
+
+/**
+ * @author Simon Jiang
+ */
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.sapphire.ExecutableElement;
+
+public interface IComponentTemplateOperation<T extends ExecutableElement> extends IComponentTemplate
```

Comment:
No need for a generic type definition, just simply it for now.  Also we can collapse these interfaces into two.  No need for two separate interfaces, just make it all IComponentTemplate